tools/mason/MasonPublish.chpl

```diff
@@ -0,0 +1,260 @@
+ * limitations under the License.
+ */
+
+
+use MasonUtils;
+use Spawn;
+use FileSystem;
+use TOML;
+use MasonEnv;
+use MasonNew;
+use MasonModify;
+use Random;
+
+/* Top Level procedure that gets called from mason.chpl that takes in arguments from command line
+Returns the help output in '-h' or '--help' exits in the arguments.
+If --dry-run is passed then it checks to see if the package is able to be published.
+Takes in the username of the package owner as an argument*/
+proc masonPublish(args: [] string) throws {
+  try! {
+    if args.size == 4 {
+      masonPublishHelp();
+      exit(0);
+    }
+    var dry = false;
+    var usernameurl = gitUrl();
+    var tail = usernameurl.find("/")-1: int;
+    var head = usernameurl.find(":")+1: int;
+    var username = usernameurl(head..tail);
+    if username == '' {
+      throw new owned MasonError('Must have remote origin set up in repository to publish');
+    }
+    else if args.size == 3 || args.size == 2{
+      for arg in args[1..] {
+        if arg == '-h' || arg == '--help' {
+          masonPublishHelp();
+          exit(0);
+        }
+        else if arg == "--dry-run" {
+          dry = true;
+        }
+      }
+      if doesGitOriginExist() {
+        if dry && args.size == 3 {
+          dryRun(username);
+        }
+        else if !dry && args.size == 3 {
+          writeln('"' + args[0]+ ' '  + args[1] + ' ' + args[2] + '  does not meet the "mason publish [options]" syntax');
+          writeln('See "mason publish -h" for more details');
+          exit(0);
+        }
+        else if args.size == 2 {
+          var usernameCheck = ('git ls-remote https://github.com/' + username + '/mason-registry').split();
+          var p = spawn(usernameCheck);
+          p.wait();
+          if p.exit_status == 0 {
+            publishPackage(username);
+          }
+          else {
+                throw new owned MasonError(username + ' is not a valid GitHub username or mason-registry is not forked \nMake sure you are connected to the internet before publishing.');
+                
+          }
+        }
+        else if args.size == 4 && dry {
+          throw new owned MasonError('Must pass your username with --dry-run');
+        }
+      }
+      else {
+        throw new owned MasonError('Must have remote origin set up in repository in order to publish.');
+      }
+    }
+    else throw new owned MasonError('Usage must meet the form "mason publish [options]"');
+  }
+  catch e: MasonError {
+    writeln(e.message());
+    masonPublishHelp();
+    exit(1);
+  }
+}
+
+/* Main Script that goes through the act of publishing the package to the mason registry.
+ Takes the package owners GitHub usrename as input will throw errors through command
+line git commands if any of the git calls fails.*/
+proc publishPackage(username: string) throws {
+  const packageLocation = here.cwd();
+  var randomInt = [1];
+  fillRandom(randomInt);
```

Comment:
Let's create a random stream and pluck a value from it, rather than using `fillRandom` here:

```chpl
var stream = makeRandomStream(int);
var uniqueDir = stream.getNext(): string;
```